### ISSUE TITLE
'Generic (Action)' devices toggle when turned on or off rather than turning on or off

### ISSUE THREAD

**Issue by MirandaPoth** [joleys/niko-home-control-II#73]
I have two external blinds that are controlled in Niko as venetianblind devices. I have generic actions set up in Niko to open or close these together. HA correctly shows when they are open or closed:
![image](https://user-images.githubusercontent.com/69429932/236171947-4f970441-7f0b-4b57-948d-939260690222.png)

I use these devices in scripts, for example at night to close all the blinds in the house:
![image](https://user-images.githubusercontent.com/69429932/236172580-12e502da-136f-4b90-ade1-4eaecce59b2a.png)

The problem is, if the blinds are already closed, they open! Similarly, if the blinds are open and the 'open external blinds' device is already on, they close. Looking at nhc2 debug messages, I can see that the device is sending 'Triggered', so it's acting like a toggle.

I think the solution is for nhc2 to check the current status of the device before triggering it. Is this possible please?

Let me know if you need any more information.

**Comment by tijsverkoyen** [comment]
@MirandaPoth Aren't the blinds exposed as well? If so: why don't you use these?
The toggles are linked to a Generic Action device? 

If so it should be fairly simple, you can alter https://github.com/joleys/niko-home-control-II/blob/master/custom_components/nhc2/entities/generic_action_basicstate.py#L45 to be:

```python
    async def async_turn_on(self):
        if not self.is_on: 
          self._device.press(self._gateway)
        self.on_change()

    async def async_turn_off(self):
        if self.is_on: 
          self._device.press(self._gateway)
        self.on_change()
```

**Comment by MirandaPoth** [comment]
@tijsverkoyen You’re right, I can use the blinds themselves individually. I have other generic actions exhibiting the same behaviour though, and lots of scripts & automations involving them! I thought it might be an easy fix. 

Thanks very much for the code snippet, I’ll give that a try. Much appreciated.

**Comment by MirandaPoth** [comment]
@tijsverkoyen That code works fine, thanks very much. Please could you put it into the main code, so that I don't have to remember to do it when there's an upgrade? No urgency.

**Comment by tijsverkoyen** [comment]
I have a PR ready: https://github.com/joleys/niko-home-control-II/pull/74
If there is no feedback from @joleys I will merge this and do a release next week.
